test/user_defined_output/log_response_headers_plugin_test.cc

```diff
@@ -0,0 +1,227 @@
+  TestResponseHeaderMapImpl headers{
+      {":status", "200"}, {"mytestheader1", "myvalue1"}, {"mytestheader2", "myvalue2"}};
+  EXPECT_TRUE(plugin->handleResponseHeaders(headers).ok());
+  std::vector<const HeaderEntry*> logged_headers = logger_ptr->getHeaderEntries();
+  EXPECT_EQ(logged_headers.size(), 2);
+  EXPECT_EQ(logged_headers[0]->key().getStringView(), "mytestheader1");
+  EXPECT_EQ(logged_headers[0]->value().getStringView(), "myvalue1");
+  EXPECT_EQ(logged_headers[1]->key().getStringView(), "mytestheader2");
+  EXPECT_EQ(logged_headers[1]->value().getStringView(), "myvalue2");
+}
+
+TEST(HandleResponseHeaders, OnlyLogsOnErrorsIfConfigured) {
+  std::unique_ptr<FakeHeaderLogger> logger = std::make_unique<FakeHeaderLogger>();
+  FakeHeaderLogger* logger_ptr = logger.get();
+  UserDefinedOutputPluginPtr plugin =
+      CreatePlugin("logging_mode:LM_SKIP_200_LEVEL_RESPONSES", std::move(logger));
+  TestResponseHeaderMapImpl headers_200{{":status", "200"}};
+  TestResponseHeaderMapImpl headers_400{{":status", "400"}};
+  TestResponseHeaderMapImpl headers_500{{":status", "500"}};
+  TestResponseHeaderMapImpl headers_100{{":status", "100"}};
+  EXPECT_TRUE(plugin->handleResponseHeaders(headers_200).ok());
+  std::vector<const HeaderEntry*> logged_headers = logger_ptr->getHeaderEntries();
+  EXPECT_TRUE(logged_headers.empty());
+
+  EXPECT_TRUE(plugin->handleResponseHeaders(headers_400).ok());
+  logged_headers = logger_ptr->getHeaderEntries();
+  EXPECT_EQ(logged_headers.size(), 1);
+  EXPECT_EQ(logged_headers[0]->key().getStringView(), ":status");
+  EXPECT_EQ(logged_headers[0]->value().getStringView(), "400");
+
+  EXPECT_TRUE(plugin->handleResponseHeaders(headers_500).ok());
+  logged_headers = logger_ptr->getHeaderEntries();
+  EXPECT_EQ(logged_headers.size(), 2);
+  EXPECT_EQ(logged_headers[1]->key().getStringView(), ":status");
+  EXPECT_EQ(logged_headers[1]->value().getStringView(), "500");
+
+  EXPECT_TRUE(plugin->handleResponseHeaders(headers_100).ok());
+  logged_headers = logger_ptr->getHeaderEntries();
+  EXPECT_EQ(logged_headers.size(), 3);
+  EXPECT_EQ(logged_headers[2]->key().getStringView(), ":status");
+  EXPECT_EQ(logged_headers[2]->value().getStringView(), "100");
+}
+
+TEST(HandleResponseHeaders, FailsWithInvalidLoggingMode) {
+  std::unique_ptr<HeaderLogger> logger = nullptr;
+  UserDefinedOutputPluginPtr plugin = CreatePlugin("", std::move(logger));
+  TestResponseHeaderMapImpl headers{};
+  EXPECT_EQ(plugin->handleResponseHeaders(headers).code(), absl::StatusCode::kInvalidArgument);
+}
+
+TEST(HandleResponseHeaders, FailsWithInvalidHeaderScope) {
+  std::unique_ptr<HeaderLogger> logger = nullptr;
+  UserDefinedOutputPluginPtr plugin = CreatePlugin("log_all_headers:true", std::move(logger));
+  TestResponseHeaderMapImpl headers{};
+  EXPECT_EQ(plugin->handleResponseHeaders(headers).code(), absl::StatusCode::kInvalidArgument);
+}
+
+TEST(HandleResponseData, ReturnsOk) {
```

Comment:
Can we add a few test cases:

- log_headers_with_name contains empty strings.
- log_headers_with_name contains duplicates.

Reply:
Done.